dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/core/ResourceMethodRegistry.java

```diff
@@ -0,0 +1,523 @@
+               @Override
+               public Method[] run() throws Exception {
+                  return clazz.getDeclaredMethods();
+               }
+            });
+         }
+      } catch (PrivilegedActionException pae) {
+
+      }
+      return methodList;
+   }
+
+   @Override
+   public void addResourceFactory(ResourceFactory rf, String base, ResourceClass resourceClass)
+   {
+      if (rf != null) rf.registered(providerFactory);
+      register(rf, base, resourceClass);
+   }
+
+   protected void register(ResourceFactory rf, String base, ResourceClass resourceClass)
+   {
+      for (ResourceMethod method : resourceClass.getResourceMethods())
+      {
+         processMethod(rf, base, method);
+      }
+      for (ResourceLocator method : resourceClass.getResourceLocators())
+      {
+         processMethod(rf, base, method);
+      }
+   }
+
+   /**
+    * Resteasy 2.x does not properly handle sub-resource and sub-resource locator
+    * endpoints with the same uri.  Resteasy 3.x does handle this properly.  In
+    * assisting customers identify this issue during an upgrade from Resteasy 2 to 3
+    * provides a waring when the situation is found.
+    */
+   public void checkAmbiguousUri()
+   {
+      for (Map.Entry<String, List<ResourceInvoker>> entry : this.root.getBounded().entrySet())
+      {
+         List<ResourceInvoker> values = entry.getValue();
+         if (values.size() > 1) {
+            int locatorCnt = 0;
+            int methodCnt = 0;
+            for(ResourceInvoker rInvoker : values)
+            {
+               if (rInvoker instanceof ResourceLocatorInvoker)
+               {
+                  locatorCnt++;
+               } else if (rInvoker instanceof ResourceMethodInvoker)
+               {
+                  methodCnt++;
+               }
+            }
+            if (methodCnt > 0 && locatorCnt > 0)
+            {
+               StringBuilder sb = new StringBuilder();
+               int cnt = values.size();
+               for (int i=0; i < cnt; i++) {
+                  ResourceInvoker exp = values.get(i);
+                  sb.append(exp.getMethod().getDeclaringClass().getName())
+                          .append(".")
+                          .append(exp.getMethod().getName());
+                  if (i < cnt-1)
+                  {
+                     sb.append(", ");
+                  }
+               }
+               LogMessages.LOGGER.uriAmbiguity(entry.getKey(), sb.toString());
+            }
+         }
+      }
+   }
+
+   protected void processMethod(ResourceFactory rf, String base, ResourceLocator method)
+   {
+      ResteasyUriBuilder builder = new ResteasyUriBuilderImpl();
+      if (base != null)
+         builder.path(base);
+      builder.path(method.getFullpath());
+      String fullpath = builder.getPath();
+      if (fullpath == null)
+         fullpath = "";
+
+      // Liberty Change Start
+      String httpMethod = "unknown";
```

Comment:
This is a hack, but it got the job done for now. What HTTP methods do we support? What about custom methods with the `X-HTTP-Method-Override` header?